diff --git a/build_detect_platform b/build_detect_platform
index d8d9ba1..a71f692 100644
--- a/build_detect_platform
+++ b/build_detect_platform
@@ -26,7 +26,7 @@ case `uname -s` in
         ;;
     Linux)
         PLATFORM=OS_LINUX
-        echo "PLATFORM_CFLAGS=-pthread -DOS_LINUX" >> build_config.mk
+        echo "PLATFORM_CFLAGS=-pthread -fPIC -DOS_LINUX" >> build_config.mk
         echo "PLATFORM_LDFLAGS=-pthread"  >> build_config.mk
         ;;
     SunOS)
diff --git a/include/leveldb/slice.h b/include/leveldb/slice.h
index 74ea8fa..135bbd7 100644
--- a/include/leveldb/slice.h
+++ b/include/leveldb/slice.h
@@ -77,7 +77,6 @@ class Slice {
             (memcmp(data_, x.data_, x.size_) == 0));
   }
 
- private:
   const char* data_;
   size_t size_;
 
