Index: include/leveldb/slice.h
===================================================================
--- include/leveldb/slice.h	(revision 47)
+++ include/leveldb/slice.h	(working copy)
@@ -77,7 +77,6 @@
             (memcmp(data_, x.data_, x.size_) == 0));
   }
 
- private:
   const char* data_;
   size_t size_;
 
Index: build_detect_platform
===================================================================
--- build_detect_platform	(revision 47)
+++ build_detect_platform	(working copy)
@@ -22,7 +22,7 @@
         ;;
     Linux)
         PLATFORM=OS_LINUX
-        echo "PLATFORM_CFLAGS=-pthread -DOS_LINUX" >> build_config.mk
+        echo "PLATFORM_CFLAGS=-pthread -fPIC -DOS_LINUX" >> build_config.mk
         echo "PLATFORM_LDFLAGS=-lpthread"  >> build_config.mk
         ;;
     SunOS)
